app/models/facility.rb

```diff
@@ -60,7 +60,7 @@ def self.cross_facility
 
   def products(type = nil)
     if type
-      super.where(type: type)
+      self.public_send(type.to_s.pluralize.underscore.to_sym)
```

Comment:
There's probably a better way to handle this, would be happy to hear suggestions if you have one.